cruise-control-core/src/main/java/com/linkedin/cruisecontrol/model/Entity.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.model;
+
+/**
+ * The abstract class for an entity. We use abstract class to force implementation to implement 
+ * {@link #hashCode()} and {@link #equals(Object)} method. 
+ */
+public abstract class Entity<G> {
```

Comment:
Javadoc for type parameter G.  I think it's @type

Reply:
It was param according to intellij.